packages/plugins/dart/flutter-freezed/src/config.ts

```diff
@@ -0,0 +1,522 @@
+  /**
+   * @name alwaysUseJsonKeyName
+   * @description Use @JsonKey(name: 'name') even if the name is already camelCased
+   * @default false
+   *
+   * @exampleMarkdown
+   * ```yml
+   * generates:
+   *   flutter_app/lib/data/models/app_models.dart
+   *     plugins:
+   *       - flutter-freezed
+   *     config:
+   *       alwaysUseJsonKeyName: true
+   *
+   * ```
+   */
+
+  alwaysUseJsonKeyName?: boolean;
+
+  /**
+   * @name assertNonNullableFields
+   * @description makes non-nullable fields optional properties in the Freezed class but uses Frezed's @Assert decorators to enforce them as required fields
+   * @default false
+   *
+   * @exampleMarkdown
+   * ```yml
+   * generates:
+   *   flutter_app/lib/data/models/app_models.dart
+   *     plugins:
+   *       - flutter-freezed
+   *     config:
+   *       assertNonNullableFields: true
+   *
+   * ```
+   */
+
+  assertNonNullableFields?: boolean; // TODO:
+
+  /**
+   * @name copyWith
+   * @description set to false to disable Freezed copyWith method helper
+   * @default null
+   *
+   * @exampleMarkdown
+   * ```yml
+   * generates:
+   *   flutter_app/lib/data/models/app_models.dart
+   *     plugins:
+   *       - flutter-freezed
+   *     config:
+   *       copyWith: false
+   * ```
+   */
+
+  copyWith?: boolean;
+
+  /**
+   * @name customDecorators
+   * @description maps GraphQL directives to freezed decorators. Arguments of the directive are passed using template strings: $1 is the first argument, $2 is the second... All `mapsToFreezedAs` values except `custom` are parsed so use the name of the directive without the `@` symbol as the key of the customDecorators. With the `custom` value, whatever you use as the key of the custom directive is used just as it is, and the arguments spread into a parenthesis ()
+   * @default {}
+   *
+   * @exampleMarkdown
+   * ```yml
+   * generates:
+   *   flutter_app/lib/data/models/app_models.dart
+   *     plugins:
+   *       - flutter-freezed
+   *     config:
+   *       customDecorators: {
+   *          'default' : {
+   *             mapsToFreezedAs: '@Default',
+   *             arguments: ['$0'],
+   *          },
+   *          'deprecated' : {
+   *             mapsToFreezedAs: '@deprecated',
+   *          },
+   *      'readonly' : {
+   *          mapsToFreezedAs: 'final',
+   *       },
+   *       }
+   *
+   * ```
+   */
+
+  customDecorators?: CustomDecorator; // TODO:
+
+  /**
+   * @name defaultUnionConstructor
+   * @description generate empty constructors for Union Types
+   * @default true
+   *
+   * @exampleMarkdown
+   * ```yml
+   * generates:
+   *   flutter_app/lib/data/models/app_models.dart
+   *     plugins:
+   *       - flutter-freezed
+   *     config:
+   *       defaultUnionConstructor: true
+   * ```
+   */
+
+  defaultUnionConstructor?: boolean;
+
+  /**
+   * @name equal
+   * @description set to false to disable Freezed equal method helper
+   * @default null
```

Comment:
shouldn't default value be a boolean?

Reply:
Yep, it should... fixed in my next push

Reply:
@charlypoly the presence or absence of those config options has a special meaning. They are used to decide which Freezed decorator to use. If even one of those options is set, then the plugin will use @Freezed() passing in the appropriate arguments to the @Freezed constructor. In their absence, the plugin fallback to using the normal @freezed decorator which accepts no configurations.

see this method

https://github.com/Parables/graphql-code-generator/blob/93e059d73e06ed98a074d4145100106e6a4f96fb/packages/plugins/dart/flutter-freezed/src/freezed-declaration-blocks/class-block.ts#L97

Reply:
@charlypoly, now using undefined as default values instead of null for optional boolean config options
This way if you don't specify an option, then it means you probably don't need it.

I can't use false as the default value as the presence of that option alone affects the generated output.

As an example, the `equal` option if present will annotate a class block using 

```ts
`@Freezed(equal: ${config.equal})`
```

while if absent will rather use this
```ts
'@freezed'
```